test/plugin/test_filter_add_insert_ids.rb

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative '../helper'
+
+require 'fluent/test/driver/filter'
+require 'fluent/plugin/filter_add_insert_ids'
+
+# Unit tests for filter_add_insert_ids plugin.
+class FilterAddInsertIdsTest < Test::Unit::TestCase
+  include Fluent::Plugin::AddInsertIdsFilter::ConfigConstants
+
+  CUSTOM_INSERT_ID_KEY = 'custom_insert_id_key'.freeze
+  APPLICATION_DEFAULT_CONFIG = ''.freeze
+  TEST_MESSAGE = 'test message for add_insert_ids plugin.'.freeze
+  INSERT_ID_KEY_CONFIG = %(
+    insert_id_key #{CUSTOM_INSERT_ID_KEY}
+  ).freeze
+
+  def setup
+    Fluent::Test.setup
+  end
+
+  def test_configure_insert_id_key
```

Comment:
It might be clearer to inline the values and split this into 2 tests:

def test_configure_insert_id_key_with_default
  d = create_driver('')
  assert_equal DEFAULT_INSERT_ID_KEY, d.instance.insert_id_key

def test_configure_insert_id_key_with_custom
  d = create_driver('insert_id_key custom_insert_id_key')
  assert_equal 'custom_insert_id_key', d.instance.insert_id_key

Reply:
The rest of our unit tests mostly follow the `test data provider` pattern. 

https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/test/plugin/base_test.rb#L65
https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/test/plugin/base_test.rb#L107

It's debatable which pattern is better. Let's discuss it and reach some consensus for future unit tests before introducing a stand-alone pattern. For this PR, I'll leave it as is.

If we do decide to switch, maybe this is a nice FixIt item to improve our tests' readability.